src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs

```diff
@@ -68,6 +68,21 @@ internal static bool IsTriggerCharacter(SourceText text, int characterPosition,
             return false;
         }
 
+        internal static bool IsTriggerCharacterOrArgumentListCharacter(SourceText text, int characterPosition, OptionSet options)
+        {
+            return IsTriggerCharacter(text, characterPosition, options) || IsArgumentListCharacter(text, characterPosition, options);
+        }
+
+        internal static bool IsArgumentListCharacter(SourceText text, int characterPosition, OptionSet options)
+        {
+            return IsArgumentListCharacter(text[characterPosition]);
+        }
+
+        internal static bool IsArgumentListCharacter(char ch)
+        {
+            return ch == '(' || ch == '[' || ch == ' ';
```

Comment:
the triggering on space makes me very very nervous.  we need to closely track how this feels (esp. with the option off).

In fact, can we actually check the option here and not have this be a trigger if the user has the option off?  that way we don't do all this work unneessarily?

Reply:
All entry points are guarded from this if the option is off (via option or experiment state).